### PR TITLE
feat(app): device Details design qa round 2 and fix card overflow btn bugs

### LINKED ISSUE #10892: 6.0 Feedback:  [Design QA] Round 2: T2 Device Details
### Overview

This is round 2 of the T2 Device Details page. First round review can be found at #10842

### Current Behavior

_No response_

### Expected Behavior

Round 2 feedback on T2 Device Details Page

[Link to view annotated frame in Figma](https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=14185%3A283354)

This is looking so good btw!

- [x]  (1) I have my viewport at 1000px, and wondering why the temp module card isn't locking into that second column?
- [x]  (2) Can we add in the different states to the toggle? I've pulled them all to the right of this screenshot in figma
- [x]  (3) Let's up the margin-bottom on .dZngDm to 1rem
- [x]  (4) Not sure if this is the appropriate place for this feedback: Noticing that when I open the menu overflow and click it again to close it doesn't close it. [Recording here.](https://www.loom.com/share/cfc065177ca149258cc7942663c65de2)
- [x]  (5) Just a wee bit too much spacing going on here, couldn't quite figure out what the culprit was. Should have 1rem (purple box height)
- [x]  (6) Could we have an 8px margin between columns so we don't have text get super close like this?
- [x]  (7) I'm not sure how things are linked, but I noticed that this overflow button isn't picking up @jerader's latest push to fix the various states of this button.
- [x]  (8) Also, can we push the open menu 4px down? currently sitting at the bottom of the menu button icon but should be a bit of space between them
- [x]  (9) Noticing this empty space here

 
Screen Shot 2022-06-23 at 3 37 09 PM
Screen Shot 2022-06-23 at 3 37 14 PM

### PR DESCRIPTION
**Pull request by jerader**
closes #10892 and #10797

# Overview

- Address design qa feedback on device details page including module card display, historical protocol run design tweaks, and fixing a bug with the pipette overflow menu, module overflow menu, and historical protocol run overflow menu not closing when clicking on the overflow button

# Changelog

- change `DeviceDetails` and associated components
- fix bug in `pipetteCard` by adding `MenuOverlay` and using the `useMenuHandleClickOutside` hook
- fix bug in `moduleCard` by adding `MenuOverlay` and using the `useMenuHandleClickOutside` hook and adding `zIndex` to `MenuList`
- fix bug in `historicalRunOverfowMenu` by adding `MenuOverlay` and using the `useMenuHandleClickOutside` hook
- add some logic for module card `minHeight` depending on viewport size and how many module cards there are

# Review requests

- review Device Details, does it match designs?
- does logic for module card `minHeight` make sense
- do the overflow menu buttons in pipette card and module card close when you click on the overflow btn?

# Risk assessment

low

### CODE REVIEW

**Review by mmencarelli — CHANGES_REQUESTED** [review]
Just a few things :) 

- [x] (1) Make this 1rem vs .5rem
Screen Shot 2022-07-08 at 9 43 35 AM

- [x] (2) The issue of the menu overflow button not closing the open menu on click has been fixed in that 1 module but is still persistent in a majority of places throughout this page. Can we fix? [Loom video](https://www.loom.com/share/6fd2f6b26a0a495ebe172042f814e180)

**Comment by jerader** [comment]
> [ ] (2) The issue of the menu overflow button not closing the open menu on click has been fixed in that 1 module but is still persistent in a majority of places throughout this page. Can we fix?

@mmencarelli , oh yeah, I see! I only fixed the issue with the pipette cards, not realizing the module cards also had the issue. I'll address that now 😄

**Review by mmencarelli — CHANGES_REQUESTED** [review]
Problem with the menu overflow button not closing the open menu still persisting on the protocol run section for me?

File: app/src/organisms/Devices/HistoricalProtocolRunOverflowMenu.tsx [hunk 1]
```diff
@@ -69,16 +69,19 @@ export function HistoricalProtocolRunOverflowMenu(
...
-          data-testid={`HistoricalProtocolRunOverflowMenu_${runId}`}
-        >
-          <MenuDropdown
-            {...props}
-            closeOverflowMenu={handleOverflowClick}
-            setShowDownloadRunLogToast={setShowDownloadRunLogToast}
-          />
-        </div>
+        <>
+          <div
```
Comment (koji) [inline app/src/organisms/Devices/HistoricalProtocolRunOverflowMenu.tsx]: Can we change `div` to `Box`?

File: app/src/organisms/Devices/HistoricalProtocolRunOverflowMenu.tsx [hunk 2]
```diff
@@ -40,19 +40,19 @@ export function HistoricalProtocolRunOverflowMenu(
...
-  const [showOverflowMenu, setShowOverflowMenu] = React.useState<boolean>(false)
+  const {
+    MenuOverlay,
+    handleOverflowClick,
+    showOverflowMenu,
+    setShowOverflowMenu,
+  } = useMenuHandleClickOutside()
   const protocolRunOverflowWrapperRef = useOnClickOutside({
     onClickOutside: () => setShowOverflowMenu(false),
   }) as React.RefObject<HTMLDivElement>
```
Comment (koji) [inline app/src/organisms/Devices/HistoricalProtocolRunOverflowMenu.tsx]: This is related to #11009

```suggestion
  const protocolRunOverflowWrapperRef = useOnClickOutside<HTMLDivElement>({
    onClickOutside: () => setShowOverflowMenu(false),
  })
```

**Review by koji — APPROVED** [review]
🦾

both overflow menus (Mel pointed out in design feedback and in her comment) work as expected.

**Review by mmencarelli — APPROVED** [review]

**Pull request merged**